Add tests to async controller and schedulers

marl0vv opened the pull request:
Добавил зависимости testcontainers
Добавил один интеграционный тест для контроллера. Если всё хорошо, то дальше примерно также остальные тесты напишу. 
В тестах можно repository использовать для получения записей из БД или тоже стоит через сервис?

marl0vv committed:
Add testcontainers

Add testcontainers dependency
Add one controller integration test to see that everything works fine

RavenVSS commented:
`В тестах можно repository использовать для получения записей из БД или тоже стоит через сервис?`  
Можно репозиторий, тут как удобнее

RavenVSS commented on src/test/java/com/glamik/webpconverter/SpringBootApplicationTest.java:
```diff
@@ -0,0 +1,37 @@
+@AutoConfigureMockMvc(addFilters = false)
+@ContextConfiguration(initializers = {SpringBootApplicationTest.Initializer.class})
+@TestPropertySource(properties = {"spring.config.location=classpath:application-properties"})
+public class SpringBootApplicationTest {
```
Можно сделать абстрактным, если это шаблон для теста  

Также убрать последние 2 аннотации, т.к. скорей всего они не нужны

RavenVSS commented on src/test/java/com/glamik/webpconverter/controller/ConverterAsyncControllerIT.java:
```diff
@@ -0,0 +1,62 @@
+                        .file(multipartFile))
+                .andExpect(status().isOk())
+                .andExpect(content().string(matchesPattern("^\"[0-9a-fA-F-]{36}\"$")))
+                .andReturn();
```
Можно вернуть объект Dto и его сравнить с эталонным через SoftAssertions либо assertj

marl0vv committed:
Add controller tests

I added controller tests but realized that these are scheduler tests. Will be fixed in a next commit.

marl0vv committed:
Add scheduler tests

RavenVSS commented on src/main/java/com/glamik/webpconverter/scheduler/ConversionTaskProcessingScheduler.java:
```diff
@@ -17,7 +17,7 @@ public class ConversionTaskProcessingScheduler {
     private final ProcessConversionTaskCommand processConversionTaskCommand;
 
-    @Scheduled(fixedDelay = 10000)
+    @Scheduled(fixedDelayString = "${processing.time.millis}")
```
По умолчанию все планировщики лучше выключать

RavenVSS commented on src/test/java/com/glamik/webpconverter/scheduler/ConversionTaskProcessingSchedulerIT.java:
```diff
@@ -0,0 +1,144 @@
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+class ConversionTaskProcessingSchedulerIT extends BaseSpringBootApplicationTest {
```
Конфиги для тестов лучше тут задать через:  
```
@TestPropertySource(properties = "process.test=10")
```
И на всякий случай добавь 
```
@DirtiesContext
```
чтобы планировщик не влиял на другие тесты

marl0vv committed:
Disable scheduler by default in tests

marl0vv committed:
Add tests for second request

marl0vv committed:
Add status get request test for non-existant id

Add test that checks that 404 will be returned if request for non-existant id.
Also required to fix this behavior in controller because it returned 200 when asked for non-existant id

marl0vv committed:
Add GET converted image request tests

marl0vv committed:
Add deletion scheduler tests

RavenVSS commented on src/main/java/com/glamik/webpconverter/scheduler/ConversionTaskProcessingScheduler.java:
```diff
@@ -5,19 +5,22 @@
 
 @Component
 @RequiredArgsConstructor
+@ConditionalOnProperty(name = "processing.time.millis")
```
nit: Можно добавить `havingValue="true"` и конфиг задавать вида `сonversion-task-processing-scheduler.enable`  

Например если у тебя где то по дефолту будет конфиг задаваться, то и планировщик будет всегда включён, а с этим конфигом можно лучше контролировать его работу
